src/middlewared/middlewared/plugins/system/product.py

```diff
@@ -77,16 +78,17 @@ def release_notes_url(self, version_str):
         If `version` is not provided, then the release notes URL will return
             a link for the currently installed version of SCALE.
         """
-        to_format = self.version_short() if version_str is None else version_str
-        to_format = to_format.split('-')[0].split('.')  # looks like ['23', '10', '0', '1']
-        len_to_format = len(to_format)
-        if len_to_format >= 2:
-            maj_vers = '.'.join(to_format[0:2])
-            base_url = f'https://www.truenas.com/docs/scale/{maj_vers}/gettingstarted/scalereleasenotes'
-            if len_to_format == 2:
-                return base_url
-            else:
-                return f'{base_url}/#{"".join(to_format)}'
+        parsed_version = parse_version_string(version_str)
+        if parsed_version is None:
+            raise CallError(f'Invalid version string specified: {version_str}')
+
+        version_split = parsed_version.split('.')
+        major_version = '.'.join(version_split[0:2])
```

Comment:
We can't do it like this. We can have API changes in minor versions and (very rarely) in hot-fix versions. This means we need to send the version string as is (i.e. `24.10.0`)